builtin/providers/aws/resource_aws_codebuild_project.go

```diff
@@ -0,0 +1,722 @@
+
+		if len(data["path"].(string)) > 0 {
+			projectArtifacts.Path = aws.String(data["path"].(string))
+		}
+	}
+
+	return projectArtifacts
+}
+
+func expandProjectEnvironment(d *schema.ResourceData) *codebuild.ProjectEnvironment {
+	configs := d.Get("environment").(*schema.Set).List()
+	projectEnv := &codebuild.ProjectEnvironment{}
+
+	envConfig := configs[0].(map[string]interface{})
+
+	if v := envConfig["compute_type"]; v != nil {
+		computeType := v.(string)
+		projectEnv.ComputeType = &computeType
+	}
+
+	if v := envConfig["image"]; v != nil {
+		image := v.(string)
+		projectEnv.Image = &image
+	}
+
+	if v := envConfig["type"]; v != nil {
+		envType := v.(string)
+		projectEnv.Type = &envType
+	}
+
+	if v := envConfig["environment_variable"]; v != nil {
+		envVariables := v.([]interface{})
+		if len(envVariables) > 0 {
+			projectEnvironmentVariables := make([]*codebuild.EnvironmentVariable, 0, len(envVariables))
+
+			for _, envVariablesConfig := range envVariables {
+				config := envVariablesConfig.(map[string]interface{})
+
+				projectEnvironmentVar := &codebuild.EnvironmentVariable{}
+
+				if v := config["name"].(string); v != "" {
+					projectEnvironmentVar.Name = &v
+				}
+
+				if v := config["value"].(string); v != "" {
+					projectEnvironmentVar.Value = &v
+				}
+
+				projectEnvironmentVariables = append(projectEnvironmentVariables, projectEnvironmentVar)
+			}
+
+			projectEnv.EnvironmentVariables = projectEnvironmentVariables
+		}
+	}
+
+	return projectEnv
+}
+
+func expandProjectSource(d *schema.ResourceData) codebuild.ProjectSource {
+	configs := d.Get("source").(*schema.Set).List()
+	projectSource := codebuild.ProjectSource{}
+
+	for _, configRaw := range configs {
+		data := configRaw.(map[string]interface{})
+
+		sourceType := data["type"].(string)
+		location := data["location"].(string)
+		buildspec := data["buildspec"].(string)
+
+		projectSource = codebuild.ProjectSource{
+			Type:      &sourceType,
+			Location:  &location,
+			Buildspec: &buildspec,
+		}
+
+		if v, ok := data["auth"]; ok {
+			if len(v.(*schema.Set).List()) > 0 {
+				auth := v.(*schema.Set).List()[0].(map[string]interface{})
+
+				projectSource.Auth = &codebuild.SourceAuth{
+					Type:     aws.String(auth["type"].(string)),
+					Resource: aws.String(auth["resource"].(string)),
+				}
+			}
+		}
+	}
+
+	return projectSource
+}
+
+func resourceAwsCodeBuildProjectRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	log.Printf("[DEBUG] CodeBuild Read Project: %s", d.Id())
+
+	resp, err := conn.BatchGetProjects(&codebuild.BatchGetProjectsInput{
+		Names: []*string{
+			aws.String(d.Id()),
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error retreiving Projects: %q", err)
+	}
+
+	// if nothing was found, then return no state
+	if len(resp.Projects) == 0 {
+		log.Printf("[INFO]: No projects were found, removing from state")
+		d.SetId("")
+		return nil
+	}
+
+	project := resp.Projects[0]
```

Comment:
Updated the Id to use the ARN not the name so the lookup here will only ever return 1 result.